File: Bank/cpp/src/Account.cpp
#include "Account.h"
#include "ITransactionRepository.h"
#include "IStatementPrinter.h"

Account::Account(std::shared_ptr<ITransactionRepository> transactionRepository, std::shared_ptr<IStatementPrinter> statementPrinter)
: transactionRepository{transactionRepository}, statementPrinter{statementPrinter}
{}

void
Account::deposit(const int amount)
{
        this->transactionRepository->addDeposit(amount);
}

void
Account::withdraw(const int amount)
{
        this->transactionRepository->addWithdrawal(amount);
}

void
Account::printStatement()
{
        std::list<Transaction> transactions = this->transactionRepository->allTransactions();
        this->statementPrinter->print(transactions);
}

File: Bank/java/src/main/java/BankKata/Account.java
package BankKata;

import java.util.List;

class Account {
    private TransactionRepository transactionRepository;
    private StatementPrinter statementPrinter;

    Account(TransactionRepository transactionRepository, StatementPrinter statementPrinter) {
        this.transactionRepository = transactionRepository;
        this.statementPrinter = statementPrinter;
    }

    void deposit(TransactionAmount transactionAmount) {
        this.transactionRepository.addDeposit(transactionAmount);
    }

    void withdraw(TransactionAmount transactionAmount) {
        this.transactionRepository.addWithdrawal(transactionAmount);
    }

    void printStatement() {
        List<Transaction> transactions = this.transactionRepository.allTransactions();
        this.statementPrinter.print(transactions);
    }
}
